### PR TITLE
@ #423 | update travis-ci

### PR DESCRIPTION
**Pull request by hieptranquoc**
connect #423 
Please check this PR! Thank you.

### CODE REVIEW

File: spec/utility_spec.rb
Comment (hoatle) [inline spec/utility_spec.rb]: should mirror the `lib` directory => move the file to `spec/teracy-dev/util_spec.rb`

File: .travis.yml
Comment (hoatle) [inline .travis.yml:13]: let's have these stages: "check style" => "test" => "deploy"

File: lib/teracy-dev/extension/manager.rb
Comment (hoatle) [inline lib/teracy-dev/extension/manager.rb]: should return immediately when the extension is not enabled

File: .rubocop.yml
Comment (hoatle) [inline .rubocop.yml:1]: follow our common standards here http://dev.teracy.org/docs/develop/style_standards.html#common-standards

File: lib/teracy-dev/extension/manager.rb
Comment (hoatle) [inline lib/teracy-dev/extension/manager.rb]: return if

File: lib/teracy-dev/location/git_synch.rb
Comment (hoatle) [inline lib/teracy-dev/location/git_synch.rb:2]: need @datphan to check on this

File: lib/teracy-dev/logging/progname_acceptor.rb
Comment (hoatle) [inline lib/teracy-dev/logging/progname_acceptor.rb]: LOG_PROGNAME must be kept here in the docs

File: lib/teracy-dev/util.rb
Comment (hoatle) [inline lib/teracy-dev/util.rb]: origin_hash

File: spec/fixture/config.json
Comment (hoatle) [inline spec/fixture/config.json]: move the util fixtures to `spec/fixtures/teracy-dev/util` directory instead

**Review by hoatle — CHANGES_REQUESTED** [review]
@hieptranquoc please check my comments
need @phuonglm and @datphan to continue reviewing

File: lib/teracy-dev/location/git_synch.rb
Comment (datphan) [inline lib/teracy-dev/location/git_synch.rb]: I don't think these lines is correct logic

Comment (datphan) [reply lib/teracy-dev/location/git_synch.rb]: exp: if `tag` is present then `check_branch` is still get called

File: lib/teracy-dev/location/git_synch.rb
Comment (datphan) [inline lib/teracy-dev/location/git_synch.rb:62]: it should have `begin`

**Comment by hieptranquoc** [comment]
I applied all comment, please check this PR again! Thank you all.

File: lib/teracy-dev/location/git_synch.rb
Comment (datphan) [inline lib/teracy-dev/location/git_synch.rb]: it should not have `': "` (space) character

File: lib/teracy-dev/location/git_synch.rb
Comment (datphan) [inline lib/teracy-dev/location/git_synch.rb]: if `check_tag` or `check_ref` return `false` then `check_branch` will still get called

File: lib/teracy-dev/util.rb
Comment (datphan) [inline lib/teracy-dev/util.rb]: should have `begin`

Comment (hieptranquoc) [reply lib/teracy-dev/util.rb]: > should have `begin`

In Ruby 2.5 or later, you can omit `begin` in `do-end` block.

Comment (hieptranquoc) [reply lib/teracy-dev/util.rb]: we will change this in the future!

**Comment by hieptranquoc** [comment]
I've updated! Please check again!

File: lib/teracy-dev/util.rb
Comment (datphan) [reply lib/teracy-dev/util.rb]: So we need to force user to use ruby >=2.5 which is not necessary, my ruby version:
```
$ ruby -v
ruby 2.4.0p0 (2016-12-24 revision 57164) [x86_64-darwin16]
```

Comment (hoatle) [reply lib/teracy-dev/util.rb]: @datphan make sure to review the latest changes from the PR instead.

**Comment by hieptranquoc** [comment]
@hoatle please check this PR!

**Comment by hoatle** [comment]
need @datphan to approve this.

File: lib/teracy-dev/extension/manager.rb
Comment (hoatle) [inline lib/teracy-dev/extension/manager.rb]: `compact!` or `ext_manifest_list = ext_manifest_list.compact`

**Review by hoatle — CHANGES_REQUESTED** [review]

**Comment by hoatle** [comment]
lots of changes, so we need to review and test this intensively before merging. still waiting for @datphan to approve this.

**Review by datphan — DISMISSED** [review]

**Review by hoatle — APPROVED** [review]
LGTM, will test more to fix bugs if any.

**Pull request merged**
